### ISSUE TITLE
Running inside wsgi flask

### ISSUE THREAD

**Issue by mrForce** [smitchell556/pdb-attach#15]
I have a flask app running through Apache, and I'm trying to use pdb_attach with it. I inserted this into the main script: 

`import pdb_attach
pdb_attach.listen(6222)`

After the server starts up, I tried to attach to it via: `sudo -u apache python3.6 -m pdb_attach 3782 6222`, but that didn't work. So, I tried running "nc localhost 6222", to see if anything was even listening on the port, and it just immediately returned. Running `sudo lsof -i -P -n | grep LISTEN` doesn't turn up anything listening on port 6222 either.

Then, I created a sample flask app, and stuck `pdb_attach.listen(6333)` at the top. Ran `flask run`, and then attached the pdb_attach to it, and it worked fine. So, it seems to be the fact that it's running through apache that is creating the issue. Do you have any advice for remedying this?

**Comment by smitchell556** [comment]
Interesting. My first thought is maybe Apache hasn't started the python process yet? What happens if you create an endpoint that loops infinitely, go to the endpoint in the browser, then connect with pdb-attach? Can you send a minimal code example including files related to the Apache set up for your project?

On a side note, I'm currently in the middle of a move and may not be able to run any code examples until Monday or Tuesday.

**Comment by smitchell556** [comment]
The problem you're having stems from the design of pdb-attach and Apache. It uses signals to listen for incoming client requests to attach to the process. Because of that I think you are going to run into a lot of problems using pdb-attach inside Apache.

Apparently Apache intercepts and ignores all signal handlers that are registered by the user program. There is a way around this documented [here](https://modwsgi.readthedocs.io/en/develop/user-guides/application-issues.html#registration-of-signal-handlers). When I applied the solution from the Apache docs I ran into another issue though. Not sure if this is a problem caused by Apache or by Flask in combination with Apache, but child threads are spawned (I assume by Apache/mod_wsgi) that the application code is run in. When `pdb_attach.listen()` tries to set a signal handler python throws an error: `ValueError: signal only works in main thread`, so pdb-attach isn't going to work for Flask + Apache/mod_wsgi in general.

I would hesitate to use pdb-attach from Apache anyways, especially if this is a production server. Anybody with login access to the machine can attach to the running process since it communicates via sockets and typically you want a web server to be pretty locked down on who can touch code running in production.

I have been working on another method for pdb-attach to work without signal handlers as signal handlers are unusable for this purpose on Windows, making pdb-attach unusable on Windows. I think it would work for your use case, but it's not finished.
